twinotter/derive.py

```diff
@@ -0,0 +1,198 @@
+
+    else:
+        raise ValueError("Can not calculate {} from dataset".format(name))
+
+
+def specific_humidity(dataset):
+
+    x_h20 = dataset.H2O_LICOR
+    q = constants.water_molecular_weight * x_h20 / (
+            constants.water_molecular_weight * x_h20 +
+            constants.dry_air_molecular_weight*(1-x_h20)
+    )
+
+    return q
+
+
+def along_track_wind(ds):
+    angle = np.arctan2(ds.U_OXTS, ds.V_OXTS)
+    magnitude = np.sqrt(ds.U_OXTS**2 + ds.V_OXTS**2)
+
+    flight_angle = np.deg2rad(ds.HDG_OXTS)
+
+    return magnitude * np.cos(angle - flight_angle)
+
+
+def combine_temperatures(nondeiced_temperature, deiced_temperature):
+    # Use the non-deiced temperature by default
+    temperature = nondeiced_temperature.copy()
+
+    # Replace with the deiced temperature where the non-deiced temperature is below
+    # freezing
+    frozen = np.where(temperature < scipy.constants.zero_Celsius)
+    temperature[frozen] = deiced_temperature[frozen]
+
+    return temperature
+
+
+def _rename_xarray(array, name):
+    array.rename(name)
+    array.attrs["standard_name"] = name
+    del array.attrs["long_name"]
+
+
+def _pint_to_xarray(quantity, ds, name):
+    """The metpy functions return pint quantities but we want a DataArray consistent
+    with the input DataSet
+
+    Args:
+        quantity (pint.Quantity):
+        ds (xarray.DataSet):
+        name (str):
+
+    Returns:
+        xarray.DataArray:
+    """
+    return xr.DataArray(
+        data=quantity.magnitude,
+        coords=ds.coords,
+        dims=ds.dims,
+        name=name,
+        attrs=dict(
+            long_name=name,
+            units=quantity.units.format_babel(),
+        ),
+        indexes=ds.indexes,
+    )
+
+
+# A dictionary mapping variables that can be calculated to the functions to calculate
+# them and arguments required as input to those functions
+available = dict(
+    air_temperature=dict(
+        function=combine_temperatures,
+        arguments=["TAT_ND_R", "TAT_DI_R"],
+    ),
+
+    air_potential_temperature=dict(
+        function=metpy.calc.potential_temperature,
+        arguments=["air_pressure", "air_temperature"],
+    ),
+
+    equivalent_potential_temperature=dict(
+        function=metpy.calc.equivalent_potential_temperature,
+        arguments=["air_pressure", "air_temperature", "dew_point_temperature"],
+    ),
+
+    humidity_mixing_ratio=dict(
+        function=metpy.calc.mixing_ratio_from_relative_humidity,
+        arguments=["relative_humidity", "air_temperature", "air_pressure"],
+    ),
+
+    relative_humidity=dict(
+        function=metpy.calc.relative_humidity_from_dewpoint,
+        arguments=["air_temperature", "dew_point_temperature"],
+    ),
+
+    virtual_potential_temperature=dict(
+        function=metpy.calc.virtual_temperature,
+        arguments=["air_potential_temperature", "humidity_mixing_ratio"],
+    ),
+)
+
+# Mapping from twin-otter variable names to CF standard names. This was generated from
```

Comment:
this is frustrating, but I guess unavoidable. It's not obvious which temperature someone would like to use so we have to assume which one to use. This will be handy for working with the `eurec4a-environment` code too. Thanks for doing this.